modules/library/metadata/src/main/java/org/geotools/factory/CategoryRegistry.java

```diff
@@ -0,0 +1,301 @@
+package org.geotools.factory;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMap.Builder;
+import org.geotools.util.PartiallyOrderedSet;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static java.util.Objects.requireNonNull;
+import static org.geotools.util.Utilities.stream;
+
+/**
+ * The category registry holds multiple instances per category. Categories are
+ * {@link Class classes} and instances are also accessible by the class they implement.
+ * Note that instances have to implement/extend the category they are filed under.
+ */
+class CategoryRegistry {
```

Comment:
Sorry misread this, my initial comment was incorrect.

For class javadocs this low level in the library we try and communicate Class, Responsibility, Collaboration information. This class is package visible rather than public so we do not expect direct access

Reply:
I didn't read the initial comment - does that mean the Javadoc needs improvement is it fine?